llms/mlx_lm/utils.py

```diff
@@ -180,6 +227,15 @@ def generate(
                 print(s[skip:], end="", flush=True)
                 skip = len(s)
 
+        if repetition_penalty > 1.0:
+            repetition_context.append(
+                token.item()
+            )  # Update repetition context after each token
+            print(repetition_context)
+            repetition_context = repetition_context[
+                -repetition_context_size:
+            ]  # Maintain the specified context size
```

Comment:
Shouldn't we be doing this above as well when we first make the repition_context?